pkg/pushsync/pushsync.go

```diff
@@ -397,19 +406,23 @@ func (ps *PushSync) pushToClosest(ctx context.Context, ch swarm.Chunk, origin bo
 
 			peer, err := nextPeer()
 			if err != nil {
+				if err == errContinue {
+					goto CONT
+				}
 				return nil, err
 			}
 
 			ps.metrics.TotalSendAttempts.Inc()
 
-			skipPeers = append(skipPeers, peer)
+			skipPeers.Add(peer)
 
 			go func() {
 				ctxd, cancel := context.WithTimeout(ctx, defaultTTL)
 				defer cancel()
 				ps.pushPeer(ctxd, resultChan, doneChan, peer, ch, origin)
 			}()
 
+		CONT:
 			// reached the limit, do not set timer to retry
 			if allowedRetries <= 0 || allowedPushes <= 0 {
 				continue
```

Comment:
If we continue here after an errContinue, the select case would just hang. This assumes a push has been initiated and we wait for the resultChan to unblock us.